h2o-r/h2o-package/R/explain.R

```diff
@@ -2401,6 +2401,8 @@ h2o.ice_plot <- function(model,
   if (h2o.getTypes(newdata[[column]])[[1]] == "string")
     stop("String columns are not supported by h2o.ice_plot.")
 
+  models_info <- .process_models_or_automl(model, newdata, require_single_model = TRUE)
+
   is_binomial <- is_binomial(model)
```

Comment:
I would suggest using the following instead. The `.process_models_or_automl` does some validation but most importantly enables memoizing/caching the models on R side.

This is not important for a use-case when a user uses only one function but if they use the whole `h2o.explain` we want to limit the number of calls to `h2o.getModel` as it is very expensive in R (no lazy properties). So the `h2o.explain` uses reference class (called `models_info`) to keep track of what models were previously "materialized" on R side so we don't have to do it again BUT it also requires the functions to be able to get this object instead of the model/automl object and work with it properly. 
```r
is_binomial <- is_binomial(models_info$get_model(models_info$model))
```

Reply:
thanks, added in c185cc1